### DISCUSSION TITLE
Enable method recognition by generic type constraint

### DISCUSSION THREAD

**Discussion by ijsgaus** [dotnet/csharplang#903] [General]
This is not possible now:
```
pubic class A  { }
public class B : A { }
public class C : A { }
public static class Ex
{
      public static TB Do<TB>(this TB tb) where TB : B  => return tb;
      public static TC Do<TC>(this TC tc) where TC : C => return tc;
}
```
But can be resolved in compile time as special naming convention.

**Comment by bondsbw** [comment]
Related:

#98 and https://github.com/dotnet/roslyn/issues/250
https://github.com/dotnet/roslyn/issues/7459
https://github.com/dotnet/roslyn/issues/6075
